_posts/2020-05-02-schedule-your-functions.md

```diff
@@ -0,0 +1,264 @@
+```
+
+## Schedule the functions
+
+After we have the environment up and running, now we can install the cron-connector and schedule our functions to run at specific intervals.
+
+### Deploy the cron-connector
+
+In order to install our `cron-connector` and schedule a function run the following command:
+
+```bash
+arkade install cron-connector
+```
+
+Wait for the `cron-connector` to be deployed:
+
+```bash
+kubectl rollout status -n openfaas deploy/sample-connector
+```
+
+## How to schedule
+
+The schedule of the function is configured through the `schedule` annotation following the normal cron syntax `* * * * *`. See [this](https://www.cyberciti.biz/faq/how-do-i-add-jobs-to-cron-under-linux-or-unix-oses/) blog post for some examples of the cron format.
+
+The cron-connector recognizes which functions should be ran on a schedule by annotation `topic` with value `cron-function`, we can add this annotation to the function's configurtion or during deployment
+
+Now lets go head and create a function which will run on a schedule.
+
+## Lockbot
+
+The `Lockbot` will run once per day at `00:00` and will check for old GitHub issues. The issue will be locked in case it is older than the days we will chose. For our example we can go with 3 months, which is roughly 90 days.
+
+### Get the function template from store
+
+The OpenFaaS framework has rich variety of language templates which you can use to write your own functions. In our example we will use Python 3 with the Debian based template `python3-flask-debian`:
+
+```bash
+faas-cli template store pull python3-flask-debian
+```
+
+### Generate the function
+
+In order to generate the function's backbone we will run the following command where you need to replace `<dockerhub_username>` with your actual username in dockerhub:
+
+```bash
+faas-cli new lockbot --lang python3-flask-debian --prefix <dockerhub_username> && mv lockbot.yml stack.yml
+```
+
+### Generate access token
+
+Go to the [personal access tokens](https://github.com/settings/tokens) page in GitHub and press the `Generate new token` button. The bot will use this to authenticate to the repository you chose.
+
+Now copy the generated token and create `auth-token` secret using the `faas-cli` where you need to replace the `<token>` with the actual copied access token:
+
+```bash
+faas-cli secret create auth-token --from-literal='<token>'
+```
+
+### Configure the function
+
+Open the function's configuration file called `stack.yml`. Append the following lines to the file:
+
+```yml
+    environment:
+      github_username: <github_username>
+      github_repository: <repository_to_watch>
+    secrets:
+      - auth-token
+    annotations:
+      topic: cron-function
+      schedule: "0 0 * * *"
+```
+
+Replace `<repository_to_watch>` with one of your own repositories and `<github_username>` with your GitHub username, which should also own the repository. The whole `stack.yml` file should look something like this:
+
+```yml
```

Comment:
Remove `<gateway_url>` -> this isn't the most efficient way to change gateway URLs and we are using the default anyway.

Reply:
Put that so when `OPENFAAS_URL` is present it will overwrite the variables in stack.yml and I am not sure what is the URL of the user, but will remove as you suggested :+1:

Reply:
this is replaced with `http://127.0.0.1:8080` :heavy_check_mark: